extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -271,32 +317,93 @@ export abstract class AzureAuth implements vscode.Disposable {
 				tenant: tenant.id,
 				resource: resource.endpoint
 			};
-
 			return this.getToken(tenant, resource, postData);
 		}
-
 		return this.handleInteractionRequired(tenant, resource);
 	}
 
+
+	/**
+	 * Gets the access token for the correct account and scope from the token cache, if the correct token doesn't exist in the token cache
+	 * (i.e. expired token, wrong scope, etc.), sends a request for a new token using the refresh token
+	 * @param accountId
+	 * @param azureResource
+	 * @returns The authentication result, including the access token
+	 */
+	public async getTokenMsal(accountId: string, azureResource: azdata.AzureResource, tenantId: string): Promise<AuthenticationResult | null> {
```

Comment:
Use undefined, not null

Reply:
The library we're using (MSAL) has the method acquireTokenSilent return either an AuthenticationResult or null though, so if I change this to undefined it shows an error